File: solutions/julia/run-length-encoding/1/run-length-encoding.jl
function encode(s)
    result = ""
    regexp = r"(.)\1*"
    for matched in eachmatch(regexp,s)
        l = length(matched.match)
        repno = (l > 1) ? string(l) : "" 
        result *= repno*matched[1]
    end
    result
end



function decode(s)
    result = ""
    regexp = r"(\d*)([^1-9])"
    for matched in eachmatch(regexp,s)
        reps = matched[1] == "" ? 1 : parse(Int, matched[1]) 
        result *= repeat(matched[2], reps)     
    end
    result    
end

File: solutions/dart/run-length-encoding/1/lib/run_length_encoding.dart
class RunLengthEncoder {
  String encode(String input){
    return input.replaceAllMapped(RegExp(r'(.)\1+'),
                   (Match m) => 
                    "${m[0]!.length}${m[1]!}");
  }

  String decode(String input){
    return input.replaceAllMapped(RegExp(r'(\d+)(.)' ),
                   (Match m) =>  
                    m[2]!* int.parse(m[1]!));
  }
}
